pkg/kubelet/kubelet_pods.go

```diff
@@ -2222,12 +2222,10 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 			} else {
 				preserveOldResourcesValue(v1.ResourceCPU, oldStatus.Resources.Requests, resources.Requests)
 			}
-			// TODO(tallclair,vinaykul,InPlacePodVerticalScaling): Investigate defaulting to actuated resources instead of allocated resources above
-			if _, exists := resources.Requests[v1.ResourceMemory]; exists {
-				// Get memory requests from actuated resources
-				if actuatedResources, found := kl.allocationManager.GetActuatedResources(pod.UID, allocatedContainer.Name); found {
-					resources.Requests[v1.ResourceMemory] = *actuatedResources.Requests.Memory()
-				}
+			if cStatus.Resources != nil && cStatus.Resources.MemoryRequest != nil {
+				resources.Limits[v1.ResourceMemory] = cStatus.Resources.MemoryRequest.DeepCopy()
```

Comment:
nil map panic here:
https://storage.googleapis.com/kubernetes-ci-logs/pr-logs/pull/132873/pull-kubernetes-e2e-kind/1956457245310980096/artifacts/kind-control-plane/kubelet.log
(skip to end of logs)

[I was looking to see if this PR was also impacted by a weird pull-kubernetes-e2e-gce cluster bringup flake, but it looks like it's this instead]

Reply:
Also fixed the unit test so it catches this now.